server/src/main/java/io/druid/segment/realtime/plumber/RealtimePlumber.java

```diff
@@ -352,9 +390,13 @@ public void doRun()
           {
             try {
               for (Pair<FireHydrant, Interval> pair : indexesToPersist) {
-                metrics.incrementRowOutputCount(persistHydrant(pair.lhs, schema, pair.rhs));
+                metrics.incrementRowOutputCount(
+                    persistHydrant(
```

Comment:
We could also just include the set of segments for the same chunk of metadata in more metadata on each of the segments.

I think there is value is storing it inside the segment as a form of lineage.

I also don't necessarily feel so strongly about it that I would be against a separate file, necessarily.  I don't think that has to be changed in this initial PR, however.  It actually unravels and creeps out the scope quite a bit 'cause it also requires us to consider the hand-off in terms of the full set of segments being handed off instead of individual segments being handed off (that is, if one of the set succeeds in handing off and the others fail, the real-time would believe that it needs to re-ingest the data).